lbrynet/lbrynet_daemon/Publisher.py

```diff
@@ -21,154 +17,30 @@ def __init__(self, session, lbry_file_manager, wallet):
-
-        def _show_result():
-            log.info(
-                "Success! Published %s --> lbry://%s txid: %s nout: %d",
-                self.file_name, self.publish_name, self.txid, self.nout
-            )
-            return defer.succeed({
-                'nout': self.nout,
-                'txid': self.txid,
-                'claim_id': self.claim_id,
-                'fee': self.fee,
-            })
-
-        self.publish_name = name
-        self.file_path = file_path
-        self.bid_amount = bid
-        self.metadata = metadata
-
-        # TODO: we cannot have this sort of code scattered throughout
-        #       our code base. Use polymorphism instead
-        if os.name == "nt":
-            file_mode = 'rb'
-        else:
-            file_mode = 'r'
-
-        d = self._check_file_path(self.file_path)
-        # TODO: ensure that we aren't leaving this resource open
-        d.addCallback(lambda _: create_lbry_file(self.session, self.lbry_file_manager,
-                                                 self.file_name, open(self.file_path, file_mode)))
-        d.addCallback(self.add_to_lbry_files)
-        d.addCallback(lambda _: self._create_sd_blob())
-        d.addCallback(lambda _: self._claim_name())
-        d.addCallback(lambda _: self.set_status())
-        d.addCallback(lambda _: self.start_reflector())
-        d.addCallbacks(
-            lambda _: _show_result(),
-            errback=log.fail(self._throw_publish_error),
-            errbackArgs=(
-                "An error occurred publishing %s to %s", self.file_name, self.publish_name)
-        )
-        return d
-
-    def start_reflector(self):
-        # TODO: is self.reflector_server unused?
-        reflector_server = random.choice(conf.settings['reflector_servers'])
-        reflector_address, reflector_port = reflector_server[0], reflector_server[1]
-        log.info("Reflecting new publication")
-        factory = reflector.ClientFactory(
-            self.session.blob_manager,
-            self.lbry_file_manager.stream_info_manager,
-            self.stream_hash,
-            self.publish_name
-        )
-        d = reactor.resolve(reflector_address)
-        d.addCallback(lambda ip: reactor.connectTCP(ip, reflector_port, factory))
-        d.addCallback(lambda _: factory.finished_deferred)
-        return d
-
-    def _check_file_path(self, file_path):
-        def check_file_threaded():
-            f = open(file_path)
-            f.close()
-            self.file_name = os.path.basename(self.file_path)
-            return True
-        return threads.deferToThread(check_file_threaded)
-
-    def set_lbry_file(self, lbry_file_downloader):
-        self.lbry_file = lbry_file_downloader
-        return defer.succeed(None)
-
-    def add_to_lbry_files(self, stream_hash):
-        self.stream_hash = stream_hash
+        file_name = os.path.basename(file_path)
+        with file_utils.get_read_handle(file_path) as read_handle:
```

Comment:
I don't think this works.  See https://docs.python.org/2.7/reference/datamodel.html#context-managers for more info, and https://docs.python.org/2.7/library/contextlib.html for some helpers.